deploy-agent/deployd/types/build.py

```diff
@@ -43,6 +43,26 @@ def __init__(self, jsonValue=None):
             self.publishInfo = jsonValue.get('publishInfo')
             self.publishDate = jsonValue.get('publishDate')
 
+    def __eq__(self, other):
```

Comment:
FYI see the warning about __hash__ here https://docs.python.org/3/reference/datamodel.html#object.__hash__

Also usually good to have an `isinstance` check so that comparing different types will return False and not an exception.

Reply:
Refactored __eq__ and added a __hash__ for both DeployGoal and Build.  Thanks.